src/rest-server/src/config/kubernetes.js

```diff
@@ -41,20 +51,32 @@ if (process.env.RBAC_IN_CLUSTER === 'false') {
     apiserverConfig.token = readFileSync(K8S_APISERVER_TOKEN_FILE, 'utf8');
   }
 } else {
-  const root = process.env.KUBERNETES_CLIENT_SERVICEACCOUNT_ROOT || '/var/run/secrets/kubernetes.io/serviceaccount/';
-  // By default, in rbac enabled k8s, the caPath and tokenPath is a fixed value. However, from the perspective of flexibility,
-  // user can custom the following 2 files' path in the future.
-  const caPath = K8S_APISERVER_CA_FILE || path.join(root, 'ca.crt');
-  const tokenPath = K8S_APISERVER_TOKEN_FILE || path.join(root, 'token');
-  const host = process.env.KUBERNETES_SERVICE_HOST;
-  const port = process.env.KUBERNETES_SERVICE_PORT;
-  apiserverConfig.uri = `https://${host}:${port}`;
-
-  try {
-    // Will be a buffer since SSL context can receive a buffer.
-    apiserverConfig.ca = readFileSync(caPath, 'utf8');
+  if (K8S_APISERVER_CA_FILE) {
+    k8s.Config.SERVICEACCOUNT_CA_PATH = K8S_APISERVER_CA_FILE;
+  }
+  if (K8S_APISERVER_TOKEN_FILE) {
     // Will be a string since http header can only receive a string.
-    apiserverConfig.token = readFileSync(tokenPath, 'utf8');
+    k8s.Config.SERVICEACCOUNT_TOKEN_PATH = K8S_APISERVER_TOKEN_FILE;
+  }
+  try {
+    const kc = new k8s.KubeConfig();
+    if (K8S_KUBECONFIG_PATH) {
+      kc.loadFromFile(K8S_KUBECONFIG_PATH);
```

Comment:
In webportal, I think there won't be a kubeconfig. So adding the env here is for local test?

Reply:
Yes

Reply:
Yes